src/components/documentViewer/documentViewerContainer.js

```diff
@@ -52,6 +53,16 @@ class DocumentViewerContainer extends Component {
     this.setState({ document });
   }
 
+  async obfuscateDocument(field) {
+    if (inIframe()) {
+      const { parentFrameConnection } = this.state;
+      const parent = await parentFrameConnection;
+      if (parent.updateCertificate) {
+        parent.updateCertificate(field);
```

Comment:
The child iframe should not tell the parent what is the next state of the certificate. It should tell parent what fields to obfuscate. Make sure the parent side is handling `handleObfuscation` by taking in a field/fields to remove and then do the necessary state update. 

```suggestion
        parent.handleObfuscation(field);
```

Reply:
parent function:  

updateCertificate(field) {
    const updatedDocument = obfuscateDocument(this.props.document, field);
    const updatedCertificate = getData(updatedDocument);
    this.renderDocument(updatedCertificate);
  }

Reply:
Can we fix the name with the parent connection? 
1. certificate is now changed to document.
2. the intention of this function is to get parent to handle obfusation by field instead of sending an updated cert back